### PR TITLE
Chat refactor 2 - split index.js

### PR DESCRIPTION
**Pull request by chrisnojima**
This attempts to split index.js into several pieces.
it's not clear this is the correct buckets yet but a good first start

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this too

File: shared/actions/chat/inbox.js [hunk 1]
```diff
@@ -552,13 +556,161 @@ function* filterSelectNext(action: Constants.InboxFilterSelectNext): SagaGenerat
...
+            Saga.put(Creators.clearMessages(selectedConversation)),
+            yield Saga.put(Creators.loadMoreMessages(selectedConversation, false)),
+          ])
+          return
+        }
+      }
+    }
+    // It is VERY important to pass the exact number of things to request here. The pagination system will
+    // return whatever number we ask for on newest messages due to its architecture so if we want only N
+    // newer items we have to explictly ask for N or it will give us messages older than onyNewerThan
```
Comment (buoyad) [inline shared/actions/chat/inbox.js]: `onyNewerThan` => `onlyNewerThan`?

File: shared/actions/chat/search.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import * as Selectors from '../../constants/selectors'
+import * as SearchConstants from '../../constants/search'
+import * as SearchCreators from '../search/creators'
+import * as Saga from '../../util/saga'
+import type {ReturnValue} from '../../constants/types/more'
+import type {TypedState} from '../../constants/reducer'
+
+const inSearchSelector = (state: TypedState) => state.chat.get('inSearch')
+
+function* _newChat(action: Constants.NewChat): Saga.SagaGenerator<any, any> {
```
Comment (buoyad) [inline shared/actions/chat/search.js]: :sheep::sheep: should this maybe go in `manage-thread.js`?

Comment (chrisnojima) [reply shared/actions/chat/search.js]: i think this stuff will move again once this is all done. the first pass at this is to logically leave this all the same and just move things around. next pass will be to simplify the logic

Comment (buoyad) [reply shared/actions/chat/search.js]: cool, sounds good

**Pull request merged**